pkgpanda/actions.py

```diff
@@ -0,0 +1,293 @@
+import collections
+import os
+import sys
+from functools import partial
+from subprocess import check_call
+
+from pkgpanda import PackageId, requests_fetcher
+from pkgpanda.exceptions import FetchError, PackageError, ValidationError
+from pkgpanda.util import (extract_tarball, if_exists, load_json, load_string,
+                           write_string)
+
+
+DCOS_TARGET_CONTENTS = """[Install]
+WantedBy=multi-user.target
+"""
+
+
+def activate_packages(install, repository, package_ids, systemd=True, block_systemd=False):
+    """Replace the active package set with package_ids.
+
+    install: pkgpanda.Install
+    repository: pkgpanda.Repository
+    package_ids: sequence of package IDs to activate
+    systemd: start/stop systemd services (default: True)
+    block_systemd: if systemd, block waiting for systemd services to come up (default: False)
+
+    """
+    assert isinstance(package_ids, collections.Sequence)
```

Comment:
>  It doesn't look like the code paths this ends up hitting particularly care (they just iterate over the package_ids given).

That was my reasoning: the code doesn't actually need a list, it just needs an iterable sequence. I saw the list assertion as over-specifying, but I can put it back if you feel otherwise.